src/pages/workspaces/workspace/useDeleteWorkspaceState.ts

```diff
@@ -35,83 +53,153 @@ export interface DeleteWorkspaceHookArgs {
 }
 
 
-export const useDeleteWorkspaceState = (hookArgs: DeleteWorkspaceHookArgs) : DeleteWorkspaceState => {
+export const useDeleteWorkspaceState = (hookArgs: DeleteWorkspaceHookArgs): DeleteWorkspaceState => {
   const [deleting, setDeleting] = useState(false)
   const [loading, setLoading] = useState(false)
-  const [apps, setApps] = useState<DeleteWorkspaceModalLeoApp[]>()
   const [collaboratorEmails, setCollaboratorEmails] = useState<string[] | null>()
   const [workspaceBucketUsageInBytes, setWorkspaceBucketUsageInBytes] = useState<number | null>()
-  const [controlledResourcesExist, setControlledResourcesExist] = useState(false)
-  const [deletableApps, nonDeletableApps] = _.partition(isResourceDeletable('app'), apps) as [DeleteWorkspaceModalLeoApp[], DeleteWorkspaceModalLeoApp[]]
+  const [deletingResources, setDeletingResources] = useState(false)
+  const [workspaceResources, setWorkspaceResources] = useState<WorkspaceResources | null>(null)
 
   const workspaceInfo: WorkspaceInfo = hookArgs.workspace.workspace
   const signal = useCancellation()
+  const checkAzureResourcesTimeout = useRef<number>()
+
+  const fetchWorkspaceResources = async (workspace: BaseWorkspace): Promise<WorkspaceResources> => {
+    const apps = isGoogleWorkspace(workspace) ?
+        await Ajax(signal).Apps.listWithoutProject({
+          role: 'creator',
+          saturnWorkspaceName: workspaceInfo.name
+        }) as DeleteWorkspaceModalLeoApp[] :
+        await Ajax(signal).Apps.listAppsV2(workspaceInfo.workspaceId) as DeleteWorkspaceModalLeoApp[]
+
+    // only v2 runtimes supported right now for azure
+    const currentRuntimesList = isAzureWorkspace(workspace) ?
+        await Ajax(signal).Runtimes.listV2WithWorkspace(workspaceInfo.workspaceId) as DeleteWorkspaceModalLeoRuntime[] : []
+
+    const [deletableApps, nonDeletableApps] = _.partition(isResourceDeletable('app'), apps) as [DeleteWorkspaceModalLeoApp[], DeleteWorkspaceModalLeoApp[]]
+    const [deletableRuntimes, nonDeletableRuntimes] = _.partition(isResourceDeletable('runtime'), currentRuntimesList) as [DeleteWorkspaceModalLeoRuntime[], DeleteWorkspaceModalLeoRuntime[]]
+    return {
+      nonDeleteableApps: nonDeletableApps,
+      deleteableApps: deletableApps,
+      apps,
+      deleteableRuntimes: deletableRuntimes,
+      nonDeleteableRuntimes: nonDeletableRuntimes,
+      runtimes: currentRuntimesList
+    }
+  }
 
   useOnMount(() => {
     const load = _.flow(
       withErrorReportingInModal('Error checking workspace resources', hookArgs.onDismiss),
       Utils.withBusyState(setLoading)
     )(async () => {
+      const appsInfo = await fetchWorkspaceResources(hookArgs.workspace)
+      setWorkspaceResources(appsInfo)
```

Comment:
This codepath now serves both the "v1" and "v2/azure" app use cases, which is why I pulled up out of the `isGoogleWorkspace` conditional below.